src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,105 @@
-import React from 'react';
 import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../react-app-env';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  commentsList, setCommentsList,
+}) => {
+  const [yourname, setYourname] = useState('');
```

Comment:
For such cases when you are collecting the data from form you can create simple custom hook which can be called for example useCommentForm.
The lgoic with getters and setters should be move there and you will simple can use it inside the main component.
It will be much more clear.


```
function useCommentForm() {
  const [yourname, setYourname] = useState('');
  const [youremail, setYouremail] = useState('');
  const [yourcomment, setYourcomment] = useState('');

  function handleNameChange(e) {
    setYourname(e.target.value);
  }

  function handleEmailChange(e) {
    setYouremail(e.target.value);
  }

  function handleCommentChange(e) {
    setYourcomment(e.target.value);
  }

  return [yourname, yourcomment, youremail, handleNameChange, handleEmailChange, handleCommentChange];
}

```
and in component whcih is using it:
`  const [yourname, yourcomment, youremail, handleNameChange, handleEmailChange, handleCommentChange] = useCommentForm();
`